third_party/blink/renderer/core/brave_session_cache.cc

```diff
@@ -76,6 +78,11 @@ blink::WebContentSettingsClient* GetContentSettingsClientFor(
   blink::WebContentSettingsClient* settings = nullptr;
   if (!context)
     return settings;
+  // Avoid blocking fingerprinting in WebUI pages.
+  auto protocol = context->GetSecurityContext().GetSecurityOrigin()->Protocol();
+  if (protocol != "https" && protocol != "http") {
```

Comment:
data HTML pages would be able to bypass this restriction, ex: `data:text/html,<script>alert('hi');</script>`. but then again, shields does not apply to such pages anyway. perhaps there is a method somewhere in brave-core that checks whether shields should apply to an origin, and we can reuse that method here?

Reply:
it'd be good to reuse such methods so that if it's wrong, there's only one place where it needs to be fixed

Reply:
Good idea -- I will look into that

Reply:
I'm going to leave this as-is for now, pending @aguscruiz's design work on shields for non-HTTP(S) URLs: https://docs.google.com/document/d/1NMOylklswutemt7P-2CHIdp1dF2S93ONP9u8dZp7Hhk/edit#heading=h.ls63kcwg56s1